### PR TITLE
Make shell scripts more interactive

### LINKED ISSUE #329: Parameterize command invocations
### Is your feature request related to a problem? Please describe.
The ability to parameterize some of these scripts would greatly improve the effectiveness of some of the commands. I'm not sure how this is best implemented, but generally being able to provide constrained customization could open up some neat possibilities

### Other issues that might benefit from this:
- #327
- #308 
- #289 

### Describe the solution you'd like
Tbd

### PR DESCRIPTION
**Pull request by adamperkowski**
## Type of Change
- [x] Refactoring

## Description
...

## Issues / other PRs related
- Resolves #329

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

**Comment by nnyyxxxx** [comment]
I will be working on this with Adam

File: tabs/applications-setup/alacritty-setup.sh [hunk 1]
```diff
@@ -3,34 +3,44 @@
...
-            pacman)
-                "$ESCALATION_TOOL" "$PACKAGER" -S --needed --noconfirm alacritty
-                ;;
-            *)
-                "$ESCALATION_TOOL" "$PACKAGER" install -y alacritty
-                ;;
-        esac
+    printf "%b" "${YELLOW}Do you want to install Alacritty? (Y/n): ${RC}"
+    read -r install_choice
+    if [ "$install_choice" != "n" ]; then
```
Comment (adamperkowski) [inline tabs/applications-setup/alacritty-setup.sh:8]:
```suggestion
    if [ "$install_choice" != "n" ] && [ "$install_choice" != "N" ]; then
```

File: tabs/applications-setup/alacritty-setup.sh [hunk 2]
```diff
@@ -3,34 +3,44 @@
...
     fi
 }
 
 setupAlacrittyConfig() {
-    printf "%b\n" "${YELLOW}Copy alacritty config files${RC}"
-    if [ -d "${HOME}/.config/alacritty" ] && [ ! -d "${HOME}/.config/alacritty-bak" ]; then
-        cp -r "${HOME}/.config/alacritty" "${HOME}/.config/alacritty-bak"
+    printf "%b" "${YELLOW}Do you want to backup existing configuration files? (Y/n): ${RC}"
+    read -r backup_choice
+    if [ "$backup_choice" != "n" ]; then
```
Comment (adamperkowski) [inline tabs/applications-setup/alacritty-setup.sh:30]:
```suggestion
    if [ "$backup_choice" != "n" ] && [ "$backup_choice" != "N" ]; then
```

**Comment by nnyyxxxx** [comment]
# WARNING: big diff incoming

**Comment by adamperkowski** [comment]
OH NO

**Comment by adamperkowski** [comment]
Not that big

**Comment by adamperkowski** [comment]
not planned

**Pull request closed without merging**